src/watertap_contrib/reflo/unit_models/zero_order/battery.py

```diff
@@ -0,0 +1,297 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import sys
+from pandas import DataFrame
+from collections import OrderedDict
+import textwrap
+
+# Import Pyomo libraries
+from pyomo.environ import (
+    ConcreteModel,
+    Var,
+    Param,
+    NonNegativeReals,
+    units as pyunits,
+    value,
+)
+from pyomo.network import Port
+from pyomo.common.config import ConfigBlock, ConfigValue, In
+
+# Import IDAES cores
+from idaes.core import FlowsheetBlock
+from idaes.core import declare_process_block_class, UnitModelBlockData
+from idaes.core.solvers import get_solver
+from idaes.core.util.tables import stream_table_dataframe_to_string
+import idaes.core.util.scaling as iscale
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_variables,
+    number_activated_constraints,
+    number_activated_blocks,
+)
+import idaes.logger as idaeslog
+
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("BatteryStorage", doc="Simple battery model")
+class BatteryStorageData(UnitModelBlockData):
```

Comment:
Not for this PR, but I noticed this is under ``unit_models`` directory, but is an energy model... We have a folder for solar models but probably should consolidate this and other solar models into an ``energy_models`` directory or something. This is fine here though at this point.

Reply:
Yeah I like that idea